source/common/http/header_map_impl.h

```diff
@@ -480,6 +481,27 @@ class RequestHeaderMapImpl final : public TypedHeaderMapImpl<RequestHeaderMap>,
   INLINE_REQ_RESP_STRING_HEADERS(DEFINE_INLINE_HEADER_STRING_FUNCS)
   INLINE_REQ_RESP_NUMERIC_HEADERS(DEFINE_INLINE_HEADER_NUMERIC_FUNCS)
 
+  absl::optional<absl::string_view> getTraceContext(absl::string_view key) const override {
+    return getTraceContext(LowerCaseString(key));
+  }
+
+  void setTraceContext(absl::string_view key, absl::string_view value) override {
+    Http::LowerCaseString low_case_key{key};
+    setCopy(low_case_key, value);
+  }
+
+  absl::optional<absl::string_view> getTraceContext(const LowerCaseString& key) const override {
+    auto result = get(key);
+    if (!result.empty()) {
+      return result[0]->value().getStringView();
+    }
+    return absl::nullopt;
+  }
+
+  void setTraceContext(const LowerCaseString& key, absl::string_view value) override {
+    setReferenceKey(key, value);
+  }
```

Comment:
We might want to just add an implicit conversion from `LowerCaseString` to `string_view`, so you don't need those overloads? Though we might want setTraceContext by reference or copy versions.

Reply:
I agree with the sentiment that this PR needs to be split up. So if necessary, this can be continued in a new PR.